File: packages/flutter/lib/src/resources/skills.dart
/// Skills resource — `/api/skills`.
library;

import 'base.dart';

class SkillsResource extends BaseResource {
  const SkillsResource(super.client);

  Future<List<Map<String, dynamic>>> list() async {
    final data = await client.request('GET', 'skills');
    if (data is List) return data.whereType<Map<String, dynamic>>().toList();
    return const [];
  }

  Future<Map<String, dynamic>> get(String id) async {
    final data = await client.request('GET', 'skills/$id');
    return (data as Map<String, dynamic>?) ?? {};
  }

  Future<Map<String, dynamic>> create(Map<String, dynamic> payload) async {
    final data = await client.request('POST', 'skills', body: payload);
    return (data as Map<String, dynamic>?) ?? {};
  }

  Future<Map<String, dynamic>> update(
      String id, Map<String, dynamic> payload) async {
    final data = await client.request('PUT', 'skills/$id', body: payload);
    return (data as Map<String, dynamic>?) ?? {};
  }

  Future<void> delete(String id) => client.request('DELETE', 'skills/$id');

  Future<Map<String, dynamic>> export({String? skillId}) async {
    final path = skillId == null ? 'skills/export' : 'skills/$skillId/export';
    final data = await client.request('GET', path);
    return (data as Map<String, dynamic>?) ?? {};
  }

  Future<Map<String, dynamic>> import_(Map<String, dynamic> payload) async {
    final data = await client.request('POST', 'skills/import', body: payload);
    return (data as Map<String, dynamic>?) ?? {};
  }
}

File: packages/python/src/tbd_agents/resources/skills.py
from __future__ import annotations

from typing import Any, Mapping

from tbd_agents.types import (
    ImportResult,
    Skill,
    SkillCreate,
    SkillExportBundle,
    SkillImportBundle,
    SkillUpdate,
)

from .base import BaseResource


class SkillsResource(BaseResource):
    def create(self, payload: SkillCreate | Mapping[str, Any]) -> Skill:
        return self._post("/skills", payload, model=Skill)

    def list(self) -> list[Skill]:
        return self._get_list("/skills", model=Skill)

    def get(self, skill_id: str) -> Skill:
        return self._get(f"/skills/{skill_id}", model=Skill)

    def update(self, skill_id: str, payload: SkillUpdate | Mapping[str, Any]) -> Skill:
        return self._put(f"/skills/{skill_id}", payload, model=Skill)

    def delete(self, skill_id: str) -> None:
        self._delete(f"/skills/{skill_id}")

    def export(self, skill_id: str | None = None) -> SkillExportBundle:
        path = "/skills/export" if skill_id is None else f"/skills/{skill_id}/export"
        return self._get(path, model=SkillExportBundle)

    def import_(self, payload: SkillImportBundle | Mapping[str, Any]) -> ImportResult:
        return self._post("/skills/import", payload, model=ImportResult)

